### PR TITLE
Add support for multi-node on CI

### PR DESCRIPTION
**Pull request by khluu**
Add script to run multi-node tests on CI.

### CODE REVIEW

File: .buildkite/test-template.j2 [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+steps:
+  {% for step in steps %}
+  {% if step.num_nodes >= 2 %}
+  - label: Multinode test
+    priority: 1000
+    agents:
+      queue: gpu_4_queue
+    commands:
+      - docker network create --subnet=192.168.10.0/24 docker-net
```
Comment (simon-mo) [inline .buildkite/test-template.j2:13]: what happens if the test breaks, will the network be deleted? if not, will this network creation fail on subsequent test?

**Review by simon-mo — COMMENTED** [review]
getting this merged as is LGTM (network need to be cleaned up though)
i slightly prefer this to be ran inside a bash script with arguments instead of infra setup for readability.

**Comment by khluu** [comment]
@simon-mo Yes using a bash script can also help set up exit trap that cleans up the Docker network upon test failure. I can migrate this over

**Review by simon-mo — APPROVED** [review]

**Comment by youkaichao** [comment]
sorry for the late review. is it already in the ci now?

**Comment by khluu** [comment]
> sorry for the late review. is it already in the ci now?

I added the multinode support to the template on `buildkite-ci`. After this PR is merged, you can start adding multi-node tests! cc: @simon-mo  for merging

**Pull request merged**